mentat/code_file_manager.py

```diff
@@ -72,6 +72,9 @@ async def write_changes_to_files(
         if not file_edits:
             return []
 
+        # Set pre-edit context in case /sample is called later
+        self.history.set_sample_diffs()
```

Comment:
I really it really only needs to be the `diff_active`. It's here so that, later, we can determine what changes were made in response to the last interaction. It requires a bit of diff algebra, which I also need to solve so I'll write it out here.

We want to do:
1. Set `diff_active` here, i.e. `git diff HEAD`
2. Mentat updates the code
3. You update the code
4. Set `diff_edit` to the diff between the current code and `diff_active`

In order to do #4, we need to call `git diff HEAD` again, and 'subtract' the outputs from #1.